File: Pillars/Pillars/Models/PillarVisuals.swift
import Foundation

struct PillarVisualColor: Codable, Hashable, Identifiable {
    let id: String
    let label: String
    let order: Int
    let isActive: Bool
}

struct PillarVisualIcon: Codable, Hashable, Identifiable {
    let id: String
    let label: String
    let defaultColorToken: String?
    let order: Int
    let isActive: Bool
}

struct PillarVisualsResponse: Codable, Hashable {
    let endpoint: String
    let source: String
    let updatedAt: Int
    let colors: [PillarVisualColor]
    let icons: [PillarVisualIcon]
}

File: admin-web/src/lib/pillar-visuals.ts
export type PillarVisualColor = {
  id: string;
  label: string;
  order: number;
  isActive: boolean;
};

export type PillarVisualIcon = {
  id: string;
  label: string;
  defaultColorToken?: string | null;
  order: number;
  isActive: boolean;
};

export type PillarVisualsRecord = {
  endpoint: string;
  source: string;
  updatedAt: number;
  colors: PillarVisualColor[];
  icons: PillarVisualIcon[];
};
